chainer_chemistry/dataset/splitters/base_splitter.py

```diff
@@ -1,14 +1,44 @@
+from chainer_chemistry.datasets import NumpyTupleDataset
+
+
 class BaseSplitter(object):
     def k_fold_split(self, dataset, k):
         raise NotImplementedError
 
+    def _split(self, dataset):
+        raise NotImplementedError
+
     def train_valid_test_split(self, dataset, frac_train=.8, frac_valid=.1,
                                frac_test=.1, seed=None, return_index=True):
-        train_inds, valid_inds, test_inds = self.split(dataset,
-                                                       frac_train=frac_train,
-                                                       frac_valid=frac_valid,
-                                                       frac_test=frac_test)
+        train_inds, valid_inds, test_inds = self._split(dataset,
+                                                        frac_train=frac_train,
+                                                        frac_valid=frac_valid,
+                                                        frac_test=frac_test)
         if return_index:
             return train_inds, valid_inds, test_inds
         else:
-            return dataset[train_inds], dataset[valid_inds], dataset[test_inds]
+            if type(dataset) == NumpyTupleDataset:
```

Comment:
use `isinstance(dataset, NumpyTupleDataset)`.

I think the different way is to introduce `converter` function as an argument, so that user can explicitly specify how to split based on `dataset` with given `indices`.

Default behavior would be...

```
def converter(dataset, indices):
    return dataset[indices]
```

and for NumpyTupleDataset, user can explicitly specify...
```
def converter_numpy_tuple_dataset(dataset, indices):
    return NumpyTupleDataset(*dataset.features[indices])
```